reference/constraints/Sequentially.rst

```diff
@@ -0,0 +1,142 @@
+Sequentially
+============
+
+This constraint allows you to apply a set of rules that should be validated
+step-by-step, allowing to interrupt the validation once the first violation is raised.
+
+As an alternative in situations ``Sequentially`` cannot solve, you may consider
+using :doc:`GroupSequence</validation/sequence_provider>` which allows more control.
+
+.. versionadded:: 5.1
+
+    The ``Sequentially`` constraint was introduced in Symfony 5.1.
+
+==========  ===================================================================
+Applies to  :ref:`property or method <validation-property-target>`
+Options     - `constraints`_
+            - `groups`_
+            - `payload`_
+Class       :class:`Symfony\\Component\\Validator\\Constraints\\Sequentially`
+Validator   :class:`Symfony\\Component\\Validator\\Constraints\\SequentiallyValidator`
+==========  ===================================================================
+
+Basic Usage
+-----------
+
+Suppose that you have a ``Place`` object with an ``$address`` property which
+must match the following requirements:
+- it's a non-blank string
+- of at least 10 chars long
+- with a specific format
+- and geolocalizable using an external service
+
+In such situations, you may encounter three issues:
+- the ``Length`` or ``Regex`` constraints may fail hard with a :class:`Symfony\\Component\\Validator\\Exception\\UnexpectedValueException`
+exception if the actual value is not a string, as enforced by ``Type``.
+- you may end with multiple error messages for the same property
+- you may perform a useless and heavy external call to geolocalize the address,
+while the format isn't valid.
+
+You can validate each of these constraints sequentially to solve these issues:
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // src/Localization/Place.php
+        namespace App\Localization;
+
+        use App\Validator\Constraints as AcmeAssert;
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class Place
+        {
+            /**
+             * @var string
+             *
+             * @Assert\Sequentially({
+             *     @Assert\NotBlank(),
+             *     @Assert\Type("string"),
+             *     @Assert\Length(min=10),
+             *     @Assert\Regex(Place::ADDRESS_REGEX),
```

Comment:
One minor, do you have a reflex which can be added? Did we really need the regex here? If yes, we should add it to the code example? 

But adding a regex is maybe distracting, could we online the pattern or is it anyway to complex?

What do you think of removing it completely from the example?

Reply:
I'd prefer to avoid adding a regex pattern. I used (but didn't show) a constant for the reason mentioned: the regex isn't important on its own and would be distracting.
I think the `@Regex` in the example deserve a purpose to demonstrate the step-by-step application of the rules, as it'll usually fail hard despite being used conjointly with `Type` without the `Sequential` constraint. But `@Length` demonstrate the same, so if you really prefer, I can remove it indeed.